apps/api/src/harness.ts

```diff
@@ -357,58 +338,137 @@ async function startServices() {
+    process.on("SIGTERM", resolve);
+  });
+
+  if (currentServices) {
+    await stopServices(currentServices);
+  }
+  watch.kill();
+}
+
+async function runProductionMode(): Promise<void> {
+  logger.section("Building TypeScript");
+  const build = execForward("api@build", "pnpm build");
+  await build.promise;
+  logger.success("Build complete");
+
+  const services = startServices();
 
   logger.info("Waiting for API on localhost:3002");
   await waitForPort(3002, "localhost");
-  logger.success("API is ready");
-
-  return {
-    api: api.promise,
-    worker: worker.promise,
-    nuqWorkers: nuqWorkers.map(w => w.promise),
-    indexWorker: indexWorker?.promise,
-  };
+  logger.success("All services ready");
+
+  await Promise.race([
+    new Promise<void>(resolve => {
+      process.on("SIGINT", resolve);
+      process.on("SIGTERM", resolve);
+    }),
+    services.api!.promise,
+    services.worker!.promise,
+    ...services.nuqWorkers.map(w => w.promise),
+    ...(services.indexWorker ? [services.indexWorker.promise] : []),
+  ]);
 }
 
-async function runCommand(command: string[], services: any) {
+async function runCommand(command: string[], services: Services) {
   logger.section(`Running: ${command.join(" ")}`);
   const cmd = execForward("command", command);
   await Promise.race([
     cmd.promise,
-    services.api,
-    services.worker,
-    ...services.nuqWorkers,
-    ...(services.indexWorker ? [services.indexWorker] : []),
+    services.api!.promise,
+    services.worker!.promise,
+    ...services.nuqWorkers.map(w => w.promise),
+    ...(services.indexWorker ? [services.indexWorker.promise] : []),
   ]);
 }
 
-async function waitForTermination(services: any) {
-  logger.info("All services running. Press Ctrl+C to stop");
-  await Promise.race([
-    new Promise<void>(resolve => {
-      process.on("SIGINT", resolve);
-      process.on("SIGTERM", resolve);
-    }),
-    services.api,
-    services.worker,
-    ...services.nuqWorkers,
-    ...(services.indexWorker ? [services.indexWorker] : []),
-  ]);
+let shuttingDown = false;
+async function gracefulShutdown() {
+  if (shuttingDown) return;
+  shuttingDown = true;
+
+  logger.section("Shutting down");
+  const terminationPromises = Array.from(childProcesses).map(terminateProcess);
+  await Promise.all(terminationPromises);
+  logger.success("All processes terminated");
 }
 
 function printUsage() {
   console.error(
     `${colors.bold}Usage:${colors.reset} pnpm harness <command...>\n`,
   );
   console.error(`${colors.bold}Special commands:${colors.reset}`);
-  console.error(`  --start        Start services and wait for termination`);
-  console.error(`  --start-built  Start services without rebuilding`);
-  console.error(`  --start-docker Start services (skip build completely)\n`);
   console.error(
-    `The harness ensures dependencies are installed, everything is built,`,
+    `  --start        Start in development mode (auto-restart on changes)`,
```

Comment:
There should still be a `--start` option that does not auto-restart on changes (my preference, sorry!)